db/disk/submission.go

```diff
@@ -323,3 +323,34 @@ func (this *backend) RemoveSubmission(assignment *model.Assignment, email string
 
     return true, nil;
 }
+
+func (this *backend) GetSubmissionResultHistory(assignment *model.Assignment, email string) ([]*model.GradingResult, error) {
+    submissions := make([]*model.GradingResult, 0);
+
+    submissionsDir := this.getUserSubmissionDir(assignment.GetCourse().GetID(), assignment.GetID(), email);
+    if !util.PathExists(submissionsDir) {
+        return nil, nil;
+    }
+
+    dirents, err := os.ReadDir(submissionsDir);
+    if err != nil {
+        return nil, fmt.Errorf("Unable to read user submissions dir '%s': '%w'.", submissionsDir, err);
+    }
+
+    if len(dirents) == 0 {
+        return nil, nil;
+    }
+
+    for _, dirent := range dirents {
+        var submission *model.GradingResult;
```

Comment:
I don't think you need this declaration, just use `:=` below.

Reply:
Resolved and noted!